domain/modelconfig/handlers/interface.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package handlers
+
+import (
+	"context"
+)
+
+// ConfigHandler amends model config as it is being saved and loaded from state.
+type ConfigHandler interface {
```

Comment:
I think this is poorly named in that it tells me nothing about what level this lives at. Is it a service layer or a state layer item. We should under no circumstances be dealing with raw values of `map[string]any`. Please stamp them out.

As a design concept I like this interface but I think we need to bolster it up a little more and make this a state layer component.

We should have a set of handlers operating on one transaction that are composed into an aggregate that is given to the services layer for persistence. The final handler should be a catch all handler that dumps into the model config table.

This allows us to get the referential integrity that we are chasing.

Reply:
I disagree it is a state layer concern. And bear in mind, as far as secrets backend is concerned, we are updating the controller db, not the model db. Let's discuss f2f.

Reply:
As discussed, handlers take set of attributes to operate on. This better manages separation of concerns.